titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/GlobalAgentClusterConstraint.java

```diff
@@ -75,31 +71,23 @@ public void prepare() {
 
     @Override
     public Result evaluate(TaskRequest taskRequest, VirtualMachineCurrentState targetVM, TaskTrackerState taskTrackerState) {
-        Pair<Boolean, String> health = evaluateHealthy(targetVM);
-        if (!health.getLeft()) {
-            return new Result(false, health.getRight());
+        if (!isHealthy(targetVM)) {
+            return UNHEALTHY;
         }
         return evaluateGpuAndCapacityTierPinning(taskRequest, targetVM);
     }
 
-    private Pair<Boolean, String> evaluateHealthy(VirtualMachineCurrentState targetVM) {
-        AgentStatus status;
-        try {
-            status = agentStatusMonitor.getStatus(getAttributeValue(targetVM, schedulerConfiguration.getInstanceAttributeName()));
-        } catch (Exception e) {
-            logger.debug("Cannot evaluate health of agent: ", e);
-            return Pair.of(false, "Unhealthy: Cannot find agent");
-        }
-        boolean healthy = status.getStatusCode() == AgentStatus.AgentStatusCode.Healthy;
-        return Pair.of(healthy, status.getStatusCode().name() + ": " + status.getDescription());
+    private boolean isHealthy(VirtualMachineCurrentState targetVM) {
+        String instanceId = getAttributeValueOrEmptyString(targetVM, schedulerConfiguration.getInstanceAttributeName());
+        return !instanceId.isEmpty() && agentStatusMonitor.isHealthy(instanceId);
```

Comment:
it would be good to have a precondition fail here in case `instanceId` is empty (another PR in the future is fine).